manifests/elasticsearch5.pp

```diff
@@ -0,0 +1,31 @@
+# Class: yumrepos::elasticsearch5
+#
+# Installs the Elasticsearch 1.6 yum repository.
+#
+class yumrepos::elasticsearch5 (
+  $elasticsearch5_url = $yumrepos::params::elasticsearch5_url,
+  $elasticsearch5_enabled = $yumrepos::params::elasticsearch5_enabled,
+  $elasticsearch5_gpgcheck = $yumrepos::params::elasticsearch5_gpgcheck,
+  $elasticsearch5_includepkgs = $yumrepos::params::elasticsearch5_includepkgs,
+  $elasticsearch5_exclude = $yumrepos::params::elasticsearch5_exclude,
+) inherits yumrepos::params {
+
+  file { '/etc/pki/rpm-gpg/GPG-KEY-elasticsearch':
```

Comment:
Since this file is also used for the older elasticsearch class, it needs to have a `!defined` check both here and in `elasticsearch16.pp`. See https://github.com/tag1consulting/puppet-yumrepos/blob/master/manifests/drush8.pp#L13 for example.